apps/state/lib/state/routes_by_service.ex

```diff
@@ -45,7 +45,7 @@ defmodule State.RoutesByService do
         []
 
       items ->
-        Enum.flat_map(items, fn {_key, routes} -> routes end)
+        Enum.uniq(Enum.flat_map(items, fn {_key, routes} -> routes end))
```

Comment:
This does work, but IIRC from when we were testing it, there were also duplicates /inside/ `routes` here. Can we 1) check whether that's true and 2) if it is true, fix that issue?

Reply:
The duplicate appears to becoming from the call to `:ets.select` on [line 34](https://github.com/mbta/api/pull/391/files#diff-877a5831dd4603e26e9a5c96801afa138e3c49738a31ab7f262e555304f7c853R34). I'm not familiar with `:ets` functions which is why this is taking me a while.

Reply:
For more context `Enum.flat_map(service_ids, fn service_id -> Enum.map(route_types, fn type -> {{{service_id, type}, :_}, [], [:"$_"]} end) end)
[{{{"service", 3}, :_}, [], [:"$_"]}]` returns `[{{{"service", 3}, :_}, [], [:"$_"]}]`. Calling `:ets_select([{{{"service", 3}, :_}, [], [:"$_"]}])` seems to be returning `[{{"service", 3}, ["route", "route"]}]`.

Reply:
@paulswartz Is there a way to force `:ets.select` to not produce lists with duplicates or do I have to remove the duplicates after they're created?

Reply:
`:ets.select` is returning the data that's in the table. In this case, the data is a list with some duplicates:

|key|value|
|-|-|
|`{"service", 3}` | `["route", "route"]`

Ideally though, we wouldn't be storing a list with duplicates in the table in the first place. To fix that, you should look for where we're calling `:ets.insert` and adjusting that data so that it's not duplicated. Once that's addressed, it might also be possible to remove the `Enum.uniq` from here.